### PR TITLE
POF-34: Index content in Algolia

### PR DESCRIPTION
**Pull request by tuube**
- Installed `strapi-hook-algolia` for indexing content to Algolia
- Added utility functions for using the hook: `utils/algolia.js`
- Added indexing to correct content type's lifecycle hooks, e.g. `api/activities/models/activity.js`
- Created custom plugin for indexing content in bulk manually: `plugins/search-indexer`

For indexing to work you need to supply two env-variables: `ALGOLIA_APPLICATION_ID` and `ALGOLIA_API_KEY`, which can be gotten from the algolia console (or from the author of this PR)

### CODE REVIEW

File: config/hook.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports =({ env }) => ({
+  settings: {
+    algolia: {
+      enabled: true,
+      applicationId: env('ALGOLIA_APPLICATION_ID'),
+      apiKey: env('ALGOLIA_API_KEY'),
+      debug: true,              // default: false
+      // prefix: 'my_own_prefix'   // default: Strapi environment (strapi.config.environment)
```
Comment (Maindii) [inline config/hook.js:8]: Can this be removed?

File: plugins/search-indexer/.gitignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Don't check auto-generated stuff into git
+coverage
+node_modules
+stats.json
+package-lock.json
```
Comment (Maindii) [inline plugins/search-indexer/.gitignore:5]: Is there a reason why this is being ignored as usually it should be commited?

File: plugins/search-indexer/services/search-indexer.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * search-indexer.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+module.exports = {
+
+};
```
Comment (Maindii) [inline plugins/search-indexer/services/search-indexer.js:11]: is this file necessary?

File: plugins/search-indexer/admin/src/translations/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import ptBR from './pt-BR.json';
+import pt from './pt.json';
+import ru from './ru.json';
+import th from './th.json';
+import tr from './tr.json';
+import uk from './uk.json';
+import vi from './vi.json';
+import zhHans from './zh-Hans.json';
+import zh from './zh.json';
+import sk from './sk.json';
```
Comment (Maindii) [inline plugins/search-indexer/admin/src/translations/index.js:22]: Could we have one empty json file so that we are not littered with a ton of empty files?

**Review by Maindii — CHANGES_REQUESTED** [review]

File: plugins/search-indexer/admin/src/translations/index.js (hunk 4 above)
Comment (tuube) [reply plugins/search-indexer/admin/src/translations/index.js:22]: This is how Strapi constructed the plugin when I created it with the Strapi CLI tool, so I wouldn't touch it.

File: plugins/search-indexer/services/search-indexer.js (hunk 3 above)
Comment (tuube) [reply plugins/search-indexer/services/search-indexer.js:11]: This is how Strapi constructed the plugin when I created it with the Strapi CLI tool, so I wouldn't touch it.

File: plugins/search-indexer/.gitignore (hunk 2 above)
Comment (tuube) [reply plugins/search-indexer/.gitignore:5]: Good point 🤔 , but this is how Strapi constructed the plugin when I created it with the Strapi CLI tool, so there's probably a reason for it.

File: api/activity-group/models/activity-group.js [hunk 5]
```diff
@@ -1,8 +1,23 @@
...
-module.exports = {};
+const contentType = "activity-group";
+
+module.exports = {
+  lifecycles: {
+    afterUpdate(result, params, data) {
+      updateInAlgolia(contentType, result);
+    },
+    afterDelete(result, params) {
+      deleteFromAlgolia(contentType, result.id);
```
[the remark this answers is not in the document]
Comment (tuube) [reply api/activity-group/models/activity-group.js:20]: It might 🤔 . Only in the case when you would need to add something else to a content type's lifecycle hooks (which these `afterDelete` and such are), it would need some code. 

I'll make an improvement and lets see.

[the remark this answers is not in the document]
Comment (tuube) [reply api/activity-group/models/activity-group.js]: How about now?

**Review by Maindii — APPROVED** [review]
